public_html/wp-content/plugins/pattern-creator/src/plugins/viewport-header-control.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { DropdownMenu, MenuGroup, MenuItem } from '@wordpress/components';
+import { select } from '@wordpress/data';
+import { store } from '@wordpress/editor';
+import { useEffect, useState } from '@wordpress/element';
+import { registerPlugin } from '@wordpress/plugins';
+
+/**
+ * Internal dependencies
+ */
+import UnitControl from '../components/unit-control';
+import usePostData from '../hooks/use-post-data';
+
+/**
+ * Module Constants
+ */
+const CONTAINER_ID = 'pattern-viewport-container';
+const META_KEY = 'wpop_viewport_width';
+const WIDTH_DEFAULTS = [
+	{
+		label: 'Narrow (320px)',
+		value: 320,
+	},
+	{
+		label: 'Default (800px)',
+		value: 800,
+	},
+	{
+		label: 'Full (1200px)',
+		value: 1200,
+	},
+];
+
+/**
+ * Removes the button container if it exists
+ */
+const maybeRemoveContainer = () => {
+	const container = document.getElementById( CONTAINER_ID );
+
+	if ( container && container.parentElement ) {
+		container.parentElement.removeChild( container );
+	}
+};
+
+/**
+ * Inserts an empty container so we can bind our ReactDOM node.
+ *
+ * @param {external:Node} btnDomRef
+ */
+const insertContainer = ( btnDomRef ) => {
+	const container = document.createElement( 'div' );
+	container.id = CONTAINER_ID;
+
+	btnDomRef.parentElement.insertBefore( container, btnDomRef );
+};
+
+/**
+ * Insert an html element to the right.
+ *
+ * @param {external:Node} newNode Element to be added
+ */
+export const insertButton = ( newNode ) => {
+	// The Gutenberg Publish Button
+	const btnDomRef = document.querySelector( '.editor-post-publish-button__button' );
+
+	if ( ! btnDomRef ) {
+		return;
+	}
+
+	// We may re-insert the same button if it state's changes
+	maybeRemoveContainer();
+
+	insertContainer( btnDomRef );
+
+	/* eslint-disable no-undef*/
```

Comment:
What's undefined here? We can pull `render` from `@wordpress/element` if that's it?

Reply:
Yeah, this is `ReactDOM`.